### PR TITLE
Consistent titles and subtitles

### PR DESCRIPTION
**Pull request by catalinred**
This fixes the HTML validation errors for #439

- Added alt="" for the decorative images.
- Everything related to `decorative-line`, `decorative-line-center`, `.alt-bg.decorative-line or `.alt-bg .decorative-line` was removed.
- All the main headings are now `.title`, with modifiers like `title-lg`, `title-alt` or `title-center`.
- The `subtitle`, seen within chapters as an adjacent to `<h1 class="title">`, share most of styles with `title`. I propose to remove the bold styling for `subtitle` as `title` is the `h1` in those chapter pages, also two "bold" lines is too much. Also, whenever `title` comes within the DOM after the `subtitle`, removed its `margin-top` for a better typography vertical rhythm.
- Remove legacy/obsolete rules like `.intro h1, .contributors h1,.methodology h1 {...}`, `.decorative-line-mobile {...}` or `.content .intro p {...}`
- The main menu pages: Table of Contents, Contributors & Methodology now have a consistent top margin, compared to the main header.
- Last but not least, I tested a lot :)

### CODE REVIEW

File: src/static/css/page.css [hunk 1]
```diff
@@ -44,7 +43,7 @@
   grid-area: index;
 }
 .index .index-box {
-  margin: 0 8px 8px;
+  margin: 20px 8px 8px;
```
Comment (catalinred) [inline src/static/css/page.css]: This top 20px matches the pages' `h1 class="title"`, so consistent top margin everywhere.

Comment (rviscomi) [reply src/static/css/page.css]: Not sure about this, I think we want all titles to be on the top of the container. I'll need to take a closer look.

Comment (catalinred) [reply src/static/css/page.css]: This comment I added is related to the main menu pages: Table of Contents, Contributors & Methodology. now have a consistent top margin, compared to the main header.

Sure, any feedback is appreciated. Let me know if you have any questions.

**Comment by catalinred** [comment]
title-border-alignment-mobile

index-alignment-and-bold-subtitle

I made the fixes, pls check out the screenshots above. Thanks for the feedback!

Also, FWIW methodology.css is not used (anymore) so we should remove it. 
Seen lots of duplicates in both page.css and methodology.css e.g. on `.index .index-box` styles.

**Pull request merged**